### PR TITLE
Error and fail the build if yarn.lock is out of date.

### PR DESCRIPTION
**Pull request by hone**
There is no `--locked` flag for yarn, this is a workaround for that. See https://github.com/heroku/heroku-buildpack-nodejs/pull/348#issuecomment-266888429 for discussion.

### CODE REVIEW

File: lib/failure.sh [hunk 1]
```diff
@@ -27,6 +27,14 @@ fail_invalid_package_json() {
   fi
 }
 
+fail_yarn_lock_outdated() {
+  local original_yarn_lock_hash=$1
+  if [ $original_yarn_lock_hash != $(yarn_lock_digest) ]; then
+    error "yarn.lock is outdated. run \`yarn install\`, commit the updated \`yarn.lock\`, and redploy"
```
Comment (tt) [inline lib/failure.sh]: "redeploy"

**Comment by hone** [comment]
Thanks @tt!

**Comment by neerfri** [comment]
:+1:

**Comment by jmorrell** [comment]
Closing in favor of https://github.com/heroku/heroku-buildpack-nodejs/pull/453

**Pull request closed without merging**